### ISSUE TITLE
Lab1R

### ISSUE THREAD

**Issue by andrefo27** [Luozirui597/CI2024_lab1#2]
As regards **instance 1**: 
I tried some executions and i noticed that the initial solution is not valid till the end of the experiment, so there is never an optimization in terms of costs.
I changed the initial solution to something always True and i noticed that some little tweak are accepted, rarely. Maybe is as you said, the change each time is not big enough.

As regards **instance 2**, nothing to say, good job.

As regards **instance 3**:
After 20 restarts, seems that the optimal solution does not improve, this is because, as far i understood, you are resetting the current solution (that could lead to the global optimum). 
It would have been interesting to take the candidate solution for restart 20 and perform further steps to verify whether it would have undergone improvements.
It's ok by the way.

As regards **instance 4**:
It would have been interesting to take the candidate solution for restart 3 and perform further steps to verify whether it would have undergone improvements.

As regards **instance 5**:
The tweak function is interesting, but there are few steps for each restart. 

As regards **instance 6**:
From the graph results that is more and more difficult to reach a better solution after a lot of restarts as the straight lines suggest for the restarts 719 (circa) and 1769 (circa) respectively.

In conclusion I can say that you have done a good job exploring various alternatives and the graphs you have plotted offer a valid assistance in the understanding of the problem.

**Comment by Luozirui597** [comment]
Thank you for your thoughtful suggestion! I'm glad to receive recognition from Darth Vader.     :D
Overall, thank you for your encouraging words and constructive feedback! 
It’s great to know that the exploration of different alternatives and the visualizations have been helpful in understanding the problem.
I’ll keep these insights in mind and try do better!
